galley/pkg/config/processing/snapshotter/analyzingdistributor.go

```diff
@@ -226,3 +237,35 @@ func (c *context) Canceled() bool {
 		return false
 	}
 }
+
+func filterByNamespaces(namespaces map[string]struct{}, messages diag.Messages) diag.Messages {
+	var msgs diag.Messages
+	for _, m := range messages {
+		if m.Origin != nil && m.Origin.Namespace() != "" {
+			if _, ok := namespaces[m.Origin.Namespace()]; !ok {
+				continue
+			}
+		}
+		msgs = append(msgs, m)
+	}
+
+	return msgs
+}
+
+func filterBySuppressions(suppressions []AnalysisSuppression, messages diag.Messages) diag.Messages {
+	var msgs diag.Messages
+	for _, m := range messages {
+		suppress := false
+		for _, s := range suppressions {
+			if m.Origin != nil && s.Code == m.Type.Code() && s.ResourceName == m.Origin.FriendlyName() {
+				scope.Analysis.Debugf("Suppressing code %s on resource %s due to suppressions list", m.Type.Code(), m.Origin.FriendlyName())
+				suppress = true
```

Comment:
Break early here? We already know the message will be suppressed at this point.

Reply:
Good point; added